### ISSUE TITLE
totally unusable in win Edge and Chrome

### ISSUE THREAD

**Issue by jiwan2004** [kanaka/noVNC#700]
We deploy Selenium docker environment for testing.  VNC server is for outputting web page UI.
websockify is proxy server
We got no UI output in Edge and Chrome
1. it works in firework.
2. it works if we use vnc windows client directly connect the VNC server port

No any error on developer console. We can establish connection but without any output.
Below is docker logs:
26/10/2016 05:44:12 Got connection from client 192.168.0.1
26/10/2016 05:44:12   other clients:
26/10/2016 05:44:12      192.168.0.1
26/10/2016 05:44:12 Normal socket connection
26/10/2016 05:44:12 incr accepted_client=5 for 192.168.0.1:55948  sock=13
26/10/2016 05:44:12 Client Protocol Version 3.8
26/10/2016 05:44:12 Protocol version sent 3.8, using 3.8
26/10/2016 05:44:12 rfbProcessClientSecurityType: executing handler for type 2
26/10/2016 05:44:12 Pixel format for client 192.168.0.1:
26/10/2016 05:44:12   32 bpp, depth 24, little endian
26/10/2016 05:44:12   true colour: max r 255 g 255 b 255, shift r 16 g 8 b 0
26/10/2016 05:44:12 no translation needed
26/10/2016 05:44:12 rfbProcessClientNormalMessage: ignoring unsupported encoding type tightPng
26/10/2016 05:44:12 Using image quality level 6 for client 192.168.0.1
26/10/2016 05:44:12 Using JPEG subsampling 0, Q79 for client 192.168.0.1
26/10/2016 05:44:12 Using compression level 9 for client 192.168.0.1
26/10/2016 05:44:12 Enabling NewFBSize protocol extension for client 192.168.0.1
26/10/2016 05:44:12 Enabling LastRect protocol extension for client 192.168.0.1
26/10/2016 05:44:12 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFFFEFE)
26/10/2016 05:44:12 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFFFECC)
26/10/2016 05:44:12 Enabling Xvp protocol extension for client 192.168.0.1
26/10/2016 05:44:12 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFFFEC8)
26/10/2016 05:44:12 rfbProcessClientNormalMessage: ignoring unsupported encoding type Enc(0xFFFFFEC7)
26/10/2016 05:44:12 Using tight encoding for client 192.168.0.1
26/10/2016 05:44:17 client 5 network rate 2552.3 KB/sec (854372.4 eff KB/sec)
26/10/2016 05:44:17 client 5 latency:  0.5 ms
26/10/2016 05:44:17 dt1: 0.0065, dt2: 0.0000 dt3: 0.0005 bytes: 16577
26/10/2016 05:44:17 link_rate: LR_UNKNOWN - 28 ms, 152 KB/s
26/10/2016 05:44:17 client useCopyRect: 192.168.0.1 -1
26/10/2016 05:44:17 client_set_net: 192.168.0.1  0.0011
26/10/2016 05:44:17 client useCopyRect: 192.168.0.1 -1
26/10/2016 05:44:17 active keyboard: turning X autorepeat off.

Below is  websockify log
58.20.82.24 - - [26/Oct/2016 20:45:54] 58.20.82.24: Plain non-SSL (ws://) WebSocket connection
58.20.82.24 - - [26/Oct/2016 20:45:54] 58.20.82.24: Version hybi-13, base64: 'False'
58.20.82.24 - - [26/Oct/2016 20:45:54] 58.20.82.24: Path: '/websockify'
58.20.82.24 - - [26/Oct/2016 20:45:54] connecting to: 127.0.0.1:5901

no any warning or error at all.
We also disable all plugin in the Chrome.

Thanks

**Comment by samhed** [comment]
Are you using vnc.html or vnc_auto.html?

> We got no UI output in Edge and Chrome

Could you elaborate further what you mean? Perhaps share a screenshot along with better explanation

> 1. it works in firework.

I assume you mean firefox?

Did you try IE? I would assume that it will behave the same as Edge, but still

**Comment by jiwan2004** [comment]
I change a computer and try again. It works for Chrome now.  I'm using win10 and VAIO Ultrabook. I guess it's graphic card problem. Does noVNC use some tech similar to  playing video? I also got some video play issue. I can play video in my Chrome but always there is hazy trail. Time frequency issue? 
  
  I think for most of users it will work. Sony ultrbook got some problem

**Comment by samhed** [comment]
> Does noVNC use some tech similar to playing video?

noVNC uses the HTML5 feature called canvas for drawing.

**Comment by jiwan2004** [comment]
it doesn't work in IE of my notebook. By the way , it's  tablet PC

**Comment by samhed** [comment]
As I have already asked, can you elaborate? It works just fine for me in IE on a Microsoft Surface device.

**Comment by jiwan2004** [comment]
![image](https://cloud.githubusercontent.com/assets/3247990/19731869/6a90555a-9bd1-11e6-8e6d-9b70f7ecf0ba.png)
Sorry for my poor english.  The one in the left is Firefox. The another one is Chrome. IE and Microsoft Edge don't work either.

**Comment by samhed** [comment]
You said earlier that it worked in Chrome?

**Comment by DirectXMan12** [comment]
Can you specify the Chrome version exhibiting the error?  This could be due to a version incompatibility.  Another thing that would be useful is if you could open up the Chrome dev tools and show the console, in case there are any errors there (press F12 on your keyboard while in Chrome)

**Comment by jiwan2004** [comment]
Dear Solly

 

   It’ not Chrome issue. It doesn’t work in IE or windows Edge. 

 

  There is no warning or error in Chrome dev tools.

 

Maybe it’s due to conflicting with some plugin. I will try to work it out.

 

Thanks.

 

Jason

 

发件人: Solly Ross [mailto:notifications@github.com] 
发送时间: 2016年10月27日 1:56
收件人: kanaka/noVNC <noVNC@noreply.github.com>
抄送: jiwan2004 <jason.wang@hiknow.com>; Author <author@noreply.github.com>
主题: Re: [kanaka/noVNC] totally unusable in win Edge and Chrome (#700)

 

Can you specify the Chrome version exhibiting the error? This could be due to a version incompatibility. Another thing that would be useful is if you could open up the Chrome dev tools and show the console, in case there are any errors there (press F12 on your keyboard while in Chrome)

**Comment by jiwan2004** [comment]
It works for Chrome now!
I didn't change anything of my computer. The only difference is I use another server(Ubuntu) and install new vncserver. 

I remember I installed vncserver with yum of Centos last time.  Now I download Tigervnc realse version and installed on Ubuntu 12.0.

I also doubt it has someting to do with  websockify .   Every thing is fine with VNC( I can connect with client viwer in Centos ) but I got unexplained  disconnetion from web socket.

No error in Chrome dev tools. So it seems  websockify close session normally.

**Comment by jiwan2004** [comment]
It works for Chrome now!
I didn't change anything of my computer. The only difference is I use another server(Ubuntu) and install new vncserver.

I remember I installed vncserver with yum of Centos last time. Now I download Tigervnc realse version and installed on Ubuntu 12.0.

I also doubt it has someting to do with websockify . Every thing is fine with VNC( I can connect with client viwer in Centos ) but I got unexplained disconnetion from web socket.

No error in Chrome dev tools.

**Comment by jiwan2004** [comment]
vnc.html has  some bugs

Edge: no side bar  
![image](https://cloud.githubusercontent.com/assets/3247990/20358000/2530ecc8-ac65-11e6-88bb-de19292130f6.png)

it reports error: RFC server is unusable

My another browser:
 I can't see mouse so I can't click anything

if I use vncauto.html,  it works for all browser

**Comment by jiwan2004** [comment]
sorry, vnc_auto.html works

**Comment by samhed** [comment]
It works fine for me on Edge.. can you copy and post the output from the JavaScript console?

**Comment by jiwan2004** [comment]
It doesn't work again!  I reinstall  the system.
vnc_auto.html works . vnc.html doesn't. No javascript error
It's not deskop browser probelm.
webscokify reports error:
101.226.33.200 - - [17/Nov/2016 21:45:04] code 404, message File not found

we use timezone of shanghai China. In our linux environment we installed Chinese fonts.
So it's problem of vnc.html.

**Comment by samhed** [comment]
You just said "it reports error: RFC server is unusable"

Could you please paste the exact message you are getting?

**Comment by patrakov** [comment]
Does it work in Chrome if you change the scaling mode using the sidebar?

**Comment by jiwan2004** [comment]
Dear kanaka

 

  What’s scaling node?

 

I will have more try to find out why .

 

Thanks

 

Jason

     

 

发件人: Alexander E. Patrakov [mailto:notifications@github.com] 
发送时间: 2016年11月22日 17:20
收件人: kanaka/noVNC <noVNC@noreply.github.com>
抄送: jiwan2004 <jason.wang@hiknow.com>; Author <author@noreply.github.com>
主题: Re: [kanaka/noVNC] totally unusable in win Edge and Chrome (#700)

 

Does it work in Chrome if you change the scaling mode using the sidebar?

**Comment by samhed** [comment]
scaling mode is a setting that you can change from the GUI

**Comment by patrakov** [comment]
![image](https://cloud.githubusercontent.com/assets/2943163/20596442/5996f330-b260-11e6-91c4-b3addb3b05d2.png)

See this screenshot - play with "Scaling Mode" and "Clip to Window", report your findings

**Comment by patrakov** [comment]
Also please make sure that you are using the latest git version of noVNC. Because "can't click because no cursor on tablet PC" looks like a bug fixed on October 20th, by pull request #619

**Comment by jiwan2004** [comment]
Sorry, I think I find the reason. I forget I used http proxy. In windows system, chrome, IE and some other browser share the proxy so all of them doen't work. The vnc.html does not work when we set The http proxy for browser.
It's strange that vnc_auto works.
Vnc.html also have some serious performace issue in Chrome.
My suggestion is to use vnc_auto.html as default page so it will not make an unnecessary trouble for newbie.
